syft/pkg/alpm_metadata.go

```diff
@@ -0,0 +1,83 @@
+package pkg
+
+import (
+	"sort"
+	"time"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/linux"
+	"github.com/scylladb/go-set/strset"
+)
+
+const AlpmDBGlob = "**/var/lib/pacman/local/**/desc"
+
+type AlpmMetadata struct {
+	BasePackage  string           `mapstructure:"base" json:"basepackage"`
+	Package      string           `mapstructure:"name" json:"package"`
+	Version      string           `mapstructure:"version" json:"version"`
+	Description  string           `mapstructure:"desc" json:"description"`
+	Architecture string           `mapstructure:"arch" json:"architecture"`
+	Size         int              `mapstructure:"size" json:"size" cyclonedx:"size"`
+	Packager     string           `mapstructure:"packager" json:"packager"`
+	License      string           `mapstructure:"license" json:"license"`
+	URL          string           `mapstructure:"url" json:"url"`
+	Validation   string           `mapstructure:"validation" json:"validation"`
+	Reason       int              `mapstructure:"reason" json:"reason"`
+	Files        []AlpmFileRecord `mapstructure:"files" json:"files"`
+	Backup       []AlpmFileRecord `mapstructure:"backup" json:"backup"`
+}
+
+// TODO: Implement mtree support
+type AlpmFileRecord struct {
+	Path         string       `mapstruture:"path" json:"path,omitempty"`
+	Type         string       `mapstructure:"type" json:"type,omitempty"`
+	UID          string       `mapstructure:"uid" json:"uid,omitempty"`
+	GID          string       `mapstructure:"gid" json:"gid,omitempty"`
+	Time         time.Time    `mapstructure:"time" json:"time,omitempty"`
+	Size         string       `mapstructure:"size" json:"size,omitempty"`
+	Md5digest    string       `mapstructure:"md5digest" json:"md5digest,omitempty"`
+	Sha256digest string       `mapstructure:"sha256digest" json:"sh256digest,omitempty"`
+	Link         string       `mapstructure:"link" json:"link,omitempty"`
+	Digest       *file.Digest `json:"digest,omitempty"`
+}
+
+// PackageURL returns the PURL for the specific Arch Linux package (see https://github.com/package-url/purl-spec)
+func (m AlpmMetadata) PackageURL(distro *linux.Release) string {
+	qualifiers := map[string]string{
+		PURLQualifierArch: m.Architecture,
+	}
+
+	if m.BasePackage != "" {
+		qualifiers[PURLQualifierUpstream] = m.BasePackage
+	}
+
+	distroID := ""
+	if distro != nil {
+		distroID = distro.ID
+	}
+
+	return packageurl.NewPackageURL(
+		"alpm",
```

Comment:
under https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst there is no pURL type for `alpm` (which is fine, this is pretty common), however, under "other candidate types to define" there is "`arch` for Arch Linux packages".

I think I like `alpm` better, but I wanted to call out that we should probably be defaulting to the candidate `arch`.

Reply:
@wagoodman do you think we should just change this over to "arch" in anticipation of the definition being added?

Reply:
that's my vote, but don't feel too strongly. I sense that `alpm` is a more accurate name and falls more in line with the same ecosystem decision points made for the existing purl types (e.g. there is `maven` not `java`), however, my default is leaning towards the initial indications in the purl spec (though it is pretty gray).

Reply:
thanks for the discussion, I'm a-ok with `alpm` 😎